src/components/tables/filetreetable.tsx

```diff
@@ -183,26 +188,49 @@ function useSelected(props: FileTreeTableProps) {
         return next;
     }, []);
 
+    const deriveNewSelection = useRef((_: boolean) => [] as string[]);
```

Comment:
Does this have to be a ref and not a simple callback?

Reply:
If I make it a callback, there's this small quirk:
![TrguiNG_GtX8lSe7qx](https://github.com/openscopeproject/TrguiNG/assets/53523617/397fd419-7a86-4d07-9a15-eebc293548e9)
I.e. after performing an action on a directory while some of its children are filtered out, the directory becomes deselected. (because the callback is changing in this case, and eslint requires us to include the callback in useEffect's dependency array).
If it's a ref, like in the PR, this issue doesn't occur:
![TrguiNG_ihIvTn8U7F](https://github.com/openscopeproject/TrguiNG/assets/53523617/aea1351b-cdf3-4579-a1ce-b3967b9b1cd2)